packages/compartment-mapper/src/link.js

```diff
@@ -434,6 +648,34 @@ export const link = (
       shouldDeferError,
       compartments,
     });
+    /** @type {ImportNowHook | undefined} */
+    let importNowHook;
+    if (
+      isSync &&
+      makeImportNowHook &&
+      compartmentDescriptor !== entryCompartmentDescriptor
+    ) {
+      if (isDynamicRequireAllowed(compartmentDescriptor)) {
+        // `parse` must be sync here
+        const syncParse = /** @type {ParseFn} */ (parse);
+        importNowHook = makeImportNowHook({
+          entryCompartmentDescriptor,
+          packageLocation: location,
+          packageName: name,
+          parse: syncParse,
+          compartments,
+        });
+      } else {
+        // it's possible for this to still be called, but
+        // the error message would be cryptic otherwise.
+        importNowHook = () => {
+          throw new Error(
+            `Dynamic require not allowed in compartment ${q(compartmentDescriptor.name)}`,
+          );
+        };
```

Comment:
I'm not sure what else to do here.

Looking for the `dynamic` flag too early causes other tests to break, and would potentially need `dynamic: true` in the policy for attenuators (see [here](https://github.com/endojs/endo/blob/8f8f5897112520ae9ae48c83b1929c575183dc01/packages/compartment-mapper/src/policy.js#L164-L166))